### PR TITLE
[MI-2139]: UI Library integration

### PR DESCRIPTION
**Pull request by sooraj-shukla**

### CODE REVIEW

File: webapp/src/containers/Rhs/rhs.scss [hunk 1]
```diff
@@ -1,10 +1,9 @@
 @import "../../styles/abstract/variables";
```
Comment (muazzamqazi) [inline webapp/src/containers/Rhs/rhs.scss:1]: Can we use path aliases here?

File: webapp/src/containers/Rhs/index.tsx [hunk 2]
```diff
@@ -182,7 +179,7 @@ const Rhs = (): JSX.Element => {
     const {isLoading: subscriptionsLoading, data: subscriptions} = getSubscriptionsState();
     const {isLoading: deletingSubscription, isError: errorInDeletingSubscription, error: deleteSubscriptionError} = getDeleteSubscriptionState();
     return (
-        <div className='rhs-content'>
+        <div className='rhs-content position-relative padding-top-15 padding-bottom-12 padding-h-12'>
```
Comment (muazzamqazi) [inline webapp/src/containers/Rhs/index.tsx:182]:
```suggestion
        <div className='rhs-content position-relative padding-v-15 padding-h-12'>
```

File: webapp/src/containers/addOrEditSubscriptions/subComponents/channelPanel.tsx [hunk 3]
```diff
@@ -103,19 +98,21 @@ const ChannelPanel = forwardRef<HTMLDivElement, ChannelPanelProps>(({
...
-            />
-            <ModalSubTitleAndError error={error}/>
+            <div className='padding-h-12 padding-v-20 wizard__body-container'>
+                <Dropdown
+                    placeholder='Select Channel'
+                    value={channel}
+                    onChange={setChannel}
+                    options={channelOptions}
+                    required={true}
+                    error={validationFailed && 'Required'}
```
Comment (muazzamqazi) [inline webapp/src/containers/addOrEditSubscriptions/subComponents/channelPanel.tsx:111]:
```suggestion
                    error={validationFailed}
```

File: webapp/src/containers/addOrEditSubscriptions/subComponents/eventsPanel.tsx [hunk 4]
```diff
@@ -48,57 +46,59 @@ const EventsPanel = forwardRef<HTMLDivElement, EventsPanelProps>(({
...
+            <div className='padding-h-12 padding-v-20 wizard__body-container'>
+                <div className='events-panel__prev-data border-radius-8 padding-v-10 padding-h-25 margin-bottom-25'>
+                    <h4 className='events-panel__prev-data-header font-14 wt-400 margin-0'>{'Channel'}</h4>
+                    <p className='events-panel__prev-data-text font-14 wt-400 margin-v-5'>
+                        {channel?.label}
+                    </p>
+                    <h4 className='events-panel__prev-data-header font-14 wt-400 margin-top-15 record-header'>{record ? 'Record' : 'Record type'}</h4>
+                    <p className='events-panel__prev-data-text font-14 wt-400 margin-v-5'>{subscriptionType === SubscriptionType.RECORD ? record : RecordTypeLabelMap[recordType]}</p>
+                </div>
+                <label className='events-panel__label font-16 margin-bottom-12 wt-400'>{'Available events:'}</label>
```
Comment (muazzamqazi) [inline webapp/src/containers/addOrEditSubscriptions/subComponents/eventsPanel.tsx:61]:
```suggestion
                <label className='events-panel__label font-16 margin-bottom-12 wt-400'>Available events:</label>
```

File: webapp/src/containers/addOrEditSubscriptions/subComponents/index.tsx [hunk 5]
```diff
@@ -241,32 +233,24 @@ const AddOrEditSubscription = ({open, close, subscriptionData}: AddOrEditSubscri
...
         }
         if (subscriptionTypePanelOpen) {
             height = subscriptionTypePanelRef.current?.offsetHeight || PanelDefaultHeights.subscriptionTypePanel;
 
             setModalDialogHeight(height);
-            // eslint-disable-next-line no-unused-expressions
-            subscriptionTypePanelRef.current?.setAttribute('style', `max-height:${height}px;overflow:auto`);
             return;
         }
         if (!subscriptionTypePanelOpen && !recordTypePanelOpen && !searchRecordsPanelOpen && !eventsPanelOpen) {
```
Comment (muazzamqazi) [inline webapp/src/containers/addOrEditSubscriptions/subComponents/index.tsx:250]:
```suggestion
        if (subscriptionTypePanelOpen || recordTypePanelOpen || searchRecordsPanelOpen || eventsPanelOpen) {
```

File: webapp/src/containers/addOrEditSubscriptions/subComponents/subscriptionTypePanel.tsx [hunk 6]
```diff
@@ -60,18 +58,20 @@ const SubscriptionTypePanel = forwardRef<HTMLDivElement, SubscriptionTypePanelPr
...
-            />
-            <ModalSubTitleAndError error={error}/>
+            <div className='padding-h-12 padding-v-20 subscription-type-panel'>
+                <Dropdown
+                    placeholder='Select Subscription Type'
+                    value={subscriptionType}
+                    onChange={(newValue) => setSubscriptionType(newValue as SubscriptionType)}
+                    options={subscriptionTypeOptions}
+                    required={true}
+                    error={validationFailed && 'Required'}
```
Comment (muazzamqazi) [inline webapp/src/containers/addOrEditSubscriptions/subComponents/subscriptionTypePanel.tsx:71]: Add `Required` to constants

**Review by muazzamqazi — APPROVED** [review]

File: webapp/src/containers/addOrEditSubscriptions/subComponents/index.tsx [hunk 7]
```diff
@@ -439,13 +416,18 @@ const AddOrEditSubscription = ({open, close, subscriptionData}: AddOrEditSubscri
...
+                    primaryBtn={{
+                        text: getResultPanelPrimaryBtnActionOrText(false) as string,
+                        onClick: getResultPanelPrimaryBtnActionOrText(true) as (() => void) | null,
+                    }}
+                    secondaryBtn={{
+                        text: 'Close',
+                        onClick: hideModal,
+                    }}
                 />
                 {false && <CircularLoader/>}
```
Comment (manojmalik20) [inline webapp/src/containers/addOrEditSubscriptions/subComponents/index.tsx:432]: If this is not needed, we should remove this.

File: webapp/src/plugin_constants/index.ts [hunk 8]
```diff
@@ -25,6 +25,10 @@ const ApiErrorIdSubscriptionsUnauthorized = 'subscriptions_not_authorized';
 const GeneralErrorMessage = 'Something went wrong';
 const GeneralErrorSubtitleForUser = 'Please contact your system administrator.';
 const GeneralErrorSubtitleForAdmin = 'Please check the server logs.';
+const SubscriptionAddedMsg = 'Subscription added successfully!';
+const SubscriptionUpdatedMsg = 'Subscription updated successfully!';
+const DeleteSubscriptionHeading = 'Confirm Delete Subscription';
```
Comment (manojmalik20) [inline webapp/src/plugin_constants/index.ts:30]:
```suggestion
const DeleteSubscriptionHeading = 'Delete Subscription';
```
This is according to the Figma. Where did you get the heading that you've written here?

File: webapp/src/containers/addOrEditSubscriptions/subComponents/eventsPanel.tsx (hunk 4 above)
Comment (sooraj-shukla) [reply webapp/src/containers/addOrEditSubscriptions/subComponents/eventsPanel.tsx:61]: Eslint won't allow this.

File: webapp/src/containers/Rhs/rhs.scss (hunk 1 above)
Comment (sooraj-shukla) [reply webapp/src/containers/Rhs/rhs.scss:1]: I guess we don't have path aliasing setup for styles in the plugins.

File: webapp/src/plugin_constants/index.ts (hunk 8 above)
Comment (sooraj-shukla) [reply webapp/src/plugin_constants/index.ts:30]: To make it consistent with the mattermost's delete confirmation dialog header.

Comment (manojmalik20) [reply webapp/src/plugin_constants/index.ts:30]: Then, shouldn't it be "Confirm Subscription Delete"?

File: webapp/src/index.tsx [hunk 9]
```diff
@@ -6,7 +6,7 @@ import {GlobalState} from 'mattermost-redux/types/store';
 // eslint-disable-next-line import/no-unresolved
 import {PluginRegistry} from 'types/mattermost-webapp';
 
-import {ServiceNowIcon} from 'mm-ui-library';
+import {ServiceNowIcon} from '@Brightscout/mm-ui-library';
```
Comment (manojmalik20) [inline webapp/src/index.tsx:9]: Should this icon be present in the UI library? Shouldn't it be specific to this plugin?

**Review by manojmalik20 — COMMENTED** [review]
Fix lint.

File: webapp/src/index.tsx (hunk 9 above)
Comment (sooraj-shukla) [reply webapp/src/index.tsx:9]: There were many other icons that were reusable, so added this one too. This won't have any effect on build size and we'll not have to include the files for adding only one icon in the plugin.

**Review by manojmalik20 — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 10]
```diff
@@ -27,11 +27,15 @@ jobs:
           node-version: 14.x
           registry-url: https://npm.pkg.github.com/
 
+      - uses: actions/checkout@v2
+
+      - name: npmrcgen
```
Comment (manojmalik20) [inline .github/workflows/ci.yml:32]:
```suggestion
      - name: Create npmrc file
```
Also, add this whole step in `release.yml` after the `checkout` step.

**Review by manojmalik20 — APPROVED** [review]

**Pull request merged**